### PR TITLE
aichat: update header ui

### PR DESCRIPTION
**Pull request by nullhook**
Resolves https://github.com/brave/brave-browser/issues/36202

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/webui/ai_chat/ai_chat_ui_page_handler.h [hunk 1]
```diff
@@ -86,6 +86,7 @@ class AIChatUIPageHandler : public ai_chat::mojom::PageHandler,
   // content::WebContentsObserver:
   void OnVisibilityChanged(content::Visibility visibility) override;
   void GetPremiumStatus(GetPremiumStatusCallback callback) override;
+  void ClosePanel() override;
```
Comment (sangwoo108) [inline browser/ui/webui/ai_chat/ai_chat_ui_page_handler.h:89]: Shouldn't this be right below SendFeedback()?

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

File: browser/ui/webui/ai_chat/ai_chat_ui_page_handler.h (hunk 1 above)
Comment (nullhook) [reply browser/ui/webui/ai_chat/ai_chat_ui_page_handler.h]: moved

**Review by mkarolin — APPROVED** [review]
`strings`++

File: components/ai_chat/resources/page/components/main/style.module.scss [hunk 2]
```diff
@@ -68,10 +85,11 @@
 }
 
 .logoTitle {
+  font-family: Poppins, sans-serif;
```
Comment (petemill) [inline components/ai_chat/resources/page/components/main/style.module.scss:88]: Should be a Nala variable?

Comment (nullhook) [reply components/ai_chat/resources/page/components/main/style.module.scss:88]: image

nala font variables come with line height and font sizes defined. i could still use them and do an override? i dunno.

Comment (petemill) [reply components/ai_chat/resources/page/components/main/style.module.scss:88]: @aguscruiz is this not one of our standard font sizes? I guess we can override just font-size if not @nullhook

Comment (aguscruiz) [reply components/ai_chat/resources/page/components/main/style.module.scss:88]: I have a style created for it in the browser tokens file. But I think that file has never been merged into the actual code you have for Nala

image

Comment (nullhook) [reply components/ai_chat/resources/page/components/main/style.module.scss]: i have updated to use a nala token and then override

File: browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc [hunk 3]
```diff
@@ -426,6 +427,14 @@ void AIChatUIPageHandler::GetPremiumStatus(GetPremiumStatusCallback callback) {
                      weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
 }
 
+void AIChatUIPageHandler::ClosePanel() {
+#if !BUILDFLAG(IS_ANDROID)
+  ai_chat::ClosePanel(web_contents());
+#else
+  ai_chat::CloseActivity(web_contents());
+#endif
```
Comment (petemill) [inline browser/ui/webui/ai_chat/ai_chat_ui_page_handler.cc:435]: very nit: we could just have both android and desktop implementations be called `ClosePanel` and then `android/ai_chat/brave_leo_settings_launch_helper.h` can implement `ClosePanel` and call `closeActivity`...

**Review by petemill — APPROVED** [review]

**Pull request merged**